### PR TITLE
S3t2 for review

### PR DESCRIPTION
**Pull request by EleanorRoberts**

### CODE REVIEW

File: src/Museum/Museum.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                $output .=              '<a href="displayCase.php?id=' . $dino->getId() . '" class="btn">More info</a>';
+                $output .=         '</div>';
+                $output .=     '</div>';
+                $output .= '</div>';
+            }
+        }
+        return $output;
+    }
+
+    public static function displayDino(Dinosaur $dino) :string
```
Comment (Roomsycakes01) [inline src/Museum/Museum.php:44]: Pull main/ have a look at my display all dinos function in s2 and update to include error messages.

File: style.scss [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+.btn:focus{
+  color: #ffffff;
+}
+
+.card-body {
+  height: 80px;
+  justify-content: space-between;
+}
+
+.row {
```
Comment (Roomsycakes01) [inline style.scss]: remove this css

File: src/Museum/Museum.php [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        $output .=                       '</div>';
+        $output .=                  '</div>';
+        $output .=                  '<a href="index.php">Go Home</a>';
+        $output .=              '</div>';
+        $output .=         '</div>';
+        $output .=     '</div>';
+        $output .= '</div>';
+        return $output;
+    }
+
```
Comment (Roomsycakes01) [inline src/Museum/Museum.php]: remove un needed whitespace

File: style.scss [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  position: relative;
+  width: 250px;
+  height: 250px;
+  overflow: hidden;
+  border: 1px solid $darkGreen;
+  background-repeat: no-repeat;
+  background-position: center top;
+  background-size: cover;
+}
+
```
Comment (Roomsycakes01) [inline style.scss]: Get a media query on the display case image/ use bootstrap for responsiveness :)

File: tests/Museum/MuseumTest.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        $this->assertEquals($expected, $result);
+    }
+
+    public function testMalformedDisplayAllDinos()
+    {
+        $this->expectException(TypeError::class);
+        $case = Museum::displayAllDinos('');
+    }
+}
```
Comment (Roomsycakes01) [inline tests/Museum/MuseumTest.php:52]: get your unit tests in!! (Not sure if this is in t1)

File: displayCase.php [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+</head>
+<body>
+<nav class="navbar sticky-top justify-content-center">
+    <img width="150px" alt="Capysaurus Logo" src="Images/CapybarasaurusLogo.gif" />
+    <h1>Capynotasaurus</h1>
+</nav>
+
+    <div class="body row g-0 m-4 justify-content-center">
+            <?php
+            echo Museum::displayDino(DinosaurHydrator::getDino($_GET['id']));
```
Comment (Roomsycakes01) [inline displayCase.php]: check if GET exists

File: src/Museum/Museum.php [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+     *
+     * @param array $dinos an array of Dinosaur objects.
+     * @return string The html to display the dinosaurs on the homepage.
+     */
+    public static function displayAllDinos(array $dinos) :string
+    {
+        $output='';
+        foreach ($dinos as $dino){
+            if($dino instanceof Dinosaur){
+                $output .= '<style> .dino'. $dino->getId().'{background-image: url(' . $dino->getImageUrl() .');} </style>';
```
Comment (Roomsycakes01) [inline src/Museum/Museum.php]: remove inline style tag as main has it as a style property.

File: tests/Museum/MuseumTest.php [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
...ttps://upload.wikimedia.org/wikipedia/commons/thumb/1/16/Maiasaura_skeleton.jpg/220px-Maiasaura_skeleton.jpg);} </style><div class="card m-4" style="width: 18rem;"><h2 class="card-title text-center mt-3">test</h2><div class="dino-img-container mx-auto dino1"></div><div class="card-body d-flex flex-row align-items-center py-5"><div class="food-type d-flex flex-row align-items-center"><img width="50px" alt="Icon to represent Herbivore" src="Icons/herbivore.png"/><p class="card-text">Herbivore</p></div><div class="button"><a href="#" class="btn">More info</a></div></div></div><style> .dino1{background-image: url(https://upload.wikimedia.org/wikipedia/commons/thumb/1/16/Maiasaura_skeleton.jpg/220px-Maiasaura_skeleton.jpg);} </style><div class="card m-4" style="width: 18rem;"><h2 class="card-title text-center mt-3">test</h2><div class="dino-img-container mx-auto dino1"></div><div class="card-body d-flex flex-row align-items-center py-5"><div class="food-type d-flex flex-row align-items-center"><img width="50px" alt="Icon to represent Herbivore" src="Icons/herbivore.png"/><p class="card-text">Herbivore</p></div><div class="button"><a href="#" class="btn">More info</a></div></div></div>';
```
Comment (Roomsycakes01) [inline tests/Museum/MuseumTest.php]: Update this after changes to Museum

File: src/Museum/Museum.php [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+        }
+        return $output;
+    }
+
+    public static function displayDino(Dinosaur $dino) :string
+    {
+        $output = '<div class="container displayCase p-4">';
+        $output .=     '<div class="row">';
+        $output .=          '<div class="col">';
+        $output .=              '<img class="w-100 img01" alt="Image of a ' . $dino->getSpecies() . '" src="' . $dino->getImageUrl() . '" />';
```
Comment (Roomsycakes01) [inline src/Museum/Museum.php:49]: Remove image of a

File: src/Museum/Museum.php [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+        $output .=          '</div>';
+        $output .=          '<div class="col-6">';
+        $output .=              '<div class="">';
+        $output .=                  '<h1 class="text-center">' . $dino->getSpecies() . '</h1>';
+        $output .=                  '<div class="d-flex justify-content-center">';
+        $output .=                      '<img width="50px" alt="Icon to represent '. $dino->getFoodType().'" src="Icons/' . $dino->getLogoUrl() . '"/>';
+        $output .=                      '<h3 class="m-0">' . $dino->getFoodType() . '</h3>';
```
Comment (Roomsycakes01) [inline src/Museum/Museum.php]: Accessibility doesn't like that you've skipped from h1 to h3.

File: src/Museum/Museum.php (hunk 7 above)
Comment (jesshamby) [reply src/Museum/Museum.php]: This was a point from Ash about having a style attribute instead of a tag so there are fewer elements in the DOM :)

**Pull request merged**